connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResourceTest.java

```diff
@@ -127,7 +143,42 @@ public void setUp() throws NoSuchMethodException {
     }
 
     @Test
-    public void testValidateConfig() throws Throwable {
+    public void testValidateConfigWithMissingConnectorClassname() throws Throwable {
+        herder.validateConnectorConfig(EasyMock.eq(partialProps));
+
+        PowerMock.expectLastCall().andAnswer(new IAnswer<ConfigInfos>() {
```

Comment:
Strictly speaking not a bug, just kind of confusing. And a reorganization of code could suddenly make this test fail despite it not currently representing realistic behavior. I guess it's a bug in the sense that the mock is not actually expressing the behavior we expect from it.